lib/pxcontext.js

```diff
@@ -51,23 +53,34 @@ class PxContext {
                 } else if ((key === '_pxvid' || key === 'pxvid') && vidRegex.test(cookies[key])) {
                     this.vid = cookies[key];
                     this.vidSource = 'vid_cookie';
+                } else if (key === 'pxcts') {
+                    this.cts = cookies[key];
                 } else if (key.match(/^_px.+$/)) {
                     this.cookies[key] = cookies[key];
                 }
             });
         }
         if (pxUtil.isGraphql(req, config)) {
             config.logger.debug('Graphql route detected');
-            this.graphqlData = this.getGraphqlDataFromBody(req.body).filter(x => x).map(
-                operation => operation && {
-                    ...operation,
-                    sensitive: pxUtil.isSensitiveGraphqlOperation(operation, config),
-                });
-            this.sensitiveGraphqlOperation = this.graphqlData.some(operation => operation && operation.sensitive);
+            this.graphqlData = this.getGraphqlDataFromBody(req.body)
+                .filter((x) => x)
+                .map(
+                    (operation) =>
+                        operation && {
+                            ...operation,
+                            sensitive: pxUtil.isSensitiveGraphqlOperation(operation, config),
+                        },
+                );
+            this.sensitiveGraphqlOperation = this.graphqlData.some((operation) => operation && operation.sensitive);
         }
         if (process.env.AWS_REGION) {
             this.serverInfoRegion = process.env.AWS_REGION;
         }
+
+        if (config.JWT_COOKIE_NAME || config.JWT_HEADER_NAME) {
+            const token = req.cookies[config.JWT_COOKIE_NAME] || req.headers[config.JWT_HEADER_NAME];
```

Comment:
We should check if this token exists before calling `extractJWTData(config, token)` -- currently will throw (and catch) an exception in the `getJWTPayload` function.